tiledb/sm/buffer/buffer.h

```diff
@@ -101,9 +175,6 @@ class Buffer {
   /** Returns the buffer data pointer at the current offset. */
   void* cur_data() const;
```

Comment:
Would this not be more appropriately placed in BufferBase(), and the _unread methods there made to reference cur_data()?

Reply:
`cur_data()` returns variously `void*` and `const void*`

Reply:
and think both can be accommodated although 
may require removing 'const' from the versions
returning non-const void *.
( void * Buffer::cur_data() const and void* PreallocatedBuffer::cur_data() const )

void * ...::cur_data() { }
const void * ...:cur_data() const {}

sort of twists my mind, but apparently both MSVC and gcc (and since you've built with clang, that too) accept it,
pattern can be seen in query/query.h,
```
  /** Returns the array. */
  const Array* array() const;

  /** Returns the array. */
  Array* array();
```

Reply:
The point *is not* that they couldn't exist from a language point of view; they can. The point *is* that the semantics of the classes don't admit it. `BufferBase` (by construction, out of convenience) is a "read-only" class so that it can be a base for `ConstBuffer`. As a result it shouldn't have non-constant public methods. (Now this is honored in the breach with casting away const with C-style casts, but you can't fix everything at once.) This is a somewhat arbitrary choice, but it's expedient for the present defect removal.

This change is far from the last word on these classes, to be sure. They're in for plenty of other modification as we get a hold on memory estimation and get more control and visibility over memory usage.